### ISSUE TITLE
Parsing datetimes using datetime.fromisoformat

### ISSUE THREAD

**Issue by OtherBarry** [Arlula/python-core-sdk#6]
Ran into another bug. I'd normally make a PR, but it requires adding a dependency which is a bit of a bigger decision.

You're currently parsing datetimes using python's standard libary `datetime.fromisoformat`. Unfortunately python is a bit backwards on this and it's not a true ISO 8601 parser. It's only designed to parse the exact formats outputted by datetime.isoformat().

[Excerpt from the python docs](https://docs.python.org/3/library/datetime.html#datetime.datetime.fromisoformat):

> **Caution** This does not support parsing arbitrary ISO 8601 strings - it is only intended as the inverse operation of `datetime.isoformat()`. A more full-featured ISO 8601 parser, `dateutil.parser.isoparse` is available in the third-party package `dateutil`.

The mentioned dateutil package is great, and is a dependecy of many major python packages, so would recommend that.

For reference, the below is the relevant part of traceback that led me to this bug. Note the specific issue in this case is that there are 5 decimal places for the seconds, and python only accepts 3 or 6.

```
~/Documents/Work/websat_obc/venv/lib/python3.9/site-packages/arlulacore/archive.py in search(self, request)
    242         else:
    243             # Break result into a list of objects
--> 244             return [SearchResult(x) for x in json.loads(response.text)]
    245 
    246     def order(self, request: OrderRequest) -> DetailedOrderResult:

~/Documents/Work/websat_obc/venv/lib/python3.9/site-packages/arlulacore/archive.py in <listcomp>(.0)
    242         else:
    243             # Break result into a list of objects
--> 244             return [SearchResult(x) for x in json.loads(response.text)]
    245 
    246     def order(self, request: OrderRequest) -> DetailedOrderResult:

~/Documents/Work/websat_obc/venv/lib/python3.9/site-packages/arlulacore/archive.py in __init__(self, data)
     64         self.platform = data["platform"]
     65         if str(data["date"]).endswith("Z"):
---> 66             self.date = datetime.fromisoformat(data["date"][:-1])
     67         self.center = CenterPoint(**data["center"])
     68         self.bounding = data["bounding"]

ValueError: Invalid isoformat string: '2021-10-18T22:38:10.61799'
```

**Comment by martejj** [comment]
Hey, thanks for pointing these issues out, should have a fix for this and the other PR soon :)

**Comment by martejj** [comment]
Fixed by #7 :)
